### PR TITLE
keepInventory setting for voidTeleport

### PR DESCRIPTION
**Pull request by MertUnverdi**

### CODE REVIEW

File: src/main/java/com/iridium/iridiumskyblock/listeners/PlayerMoveListener.java [hunk 1]
```diff
@@ -31,17 +31,22 @@ public void onPlayerMove(PlayerMoveEvent event) {
...
+                final User user = User.getUser(player);
                 final Island island = islandManager.getIslandViaLocation(location);
                 final World world = location.getWorld();
                 if (world == null) return;
 
                 if (island != null) {
-                    if (world.getName().equals(islandManager.getWorld().getName()))
+                    if (world.getName().equals(islandManager.getWorld().getName())) {
+                        if (!config.keepInventoryOnVoid && (island == user.getIsland() || island.getCoop().contains(user.getIsland().getId())))
+                            player.getInventory().clear();
```
Comment (IridiumLLC) [inline src/main/java/com/iridium/iridiumskyblock/listeners/PlayerMoveListener.java:42]: This wont clear their armor... I think

File: src/main/java/com/iridium/iridiumskyblock/listeners/PlayerMoveListener.java [hunk 2]
```diff
@@ -31,17 +31,22 @@ public void onPlayerMove(PlayerMoveEvent event) {
...
             if (location.getY() < 0 && config.voidTeleport) {
+                final User user = User.getUser(player);
                 final Island island = islandManager.getIslandViaLocation(location);
                 final World world = location.getWorld();
                 if (world == null) return;
 
                 if (island != null) {
-                    if (world.getName().equals(islandManager.getWorld().getName()))
+                    if (world.getName().equals(islandManager.getWorld().getName())) {
+                        if (!config.keepInventoryOnVoid && (island == user.getIsland() || island.getCoop().contains(user.getIsland().getId())))
```
Comment (IridiumLLC) [inline src/main/java/com/iridium/iridiumskyblock/listeners/PlayerMoveListener.java:41]: Does it matter if they arnt on their island though? If they fall into the void on any island shouldnt they lose their items?

File: src/main/java/com/iridium/iridiumskyblock/listeners/PlayerMoveListener.java [hunk 3]
```diff
@@ -31,17 +31,22 @@ public void onPlayerMove(PlayerMoveEvent event) {
...
+                            player.getInventory().clear();
                         island.teleportHome(player);
-                    else
+                    } else {
+                        if (!config.keepInventoryOnVoid && (island == user.getIsland() || island.getCoop().contains(user.getIsland().getId())))
```
Comment (IridiumLLC) [inline src/main/java/com/iridium/iridiumskyblock/listeners/PlayerMoveListener.java:45]: Why are we doing this code twice in both the if and the else? we should move it outside the statement instead.

(Unless im blind and they are different)

**Review by IridiumLLC — CHANGES_REQUESTED** [review]

File: src/main/java/com/iridium/iridiumskyblock/listeners/PlayerMoveListener.java (hunk 1 above)
Comment (MertUnverdi) [reply src/main/java/com/iridium/iridiumskyblock/listeners/PlayerMoveListener.java:42]: This suggested by CW_UK and i ask to him for about armors he said  should be delete armors tho

Comment (IridiumLLC) [reply src/main/java/com/iridium/iridiumskyblock/listeners/PlayerMoveListener.java:44]: Sorry, your misunderstanding me. I also think it should clear their armor.

But I think player.getInventory().clear() only clears their inventory, not their armor (I may be wrong)

Comment (MertUnverdi) [reply src/main/java/com/iridium/iridiumskyblock/listeners/PlayerMoveListener.java:44]: Sorry for my bad :(
Also player.getInventory().clear for all
And player.getInventory().getContents() for envanter slots

Comment (MertUnverdi) [reply src/main/java/com/iridium/iridiumskyblock/listeners/PlayerMoveListener.java:44]: i'll change to back when i at home thanks for comments <3

**Pull request closed without merging**